example-mlem-get-started/code/src/requirements.txt

```diff
@@ -0,0 +1,8 @@
+pandas
+scikit-learn
+scipy
+dvc[azure,s3]
```

Comment:
installing DVC like this will contradict the devcontainer that installs it globally - btw, why do we need it?

Reply:
We need DVC here because in one branch (`dvc`) we storing models with DVC. We don't cover this in GS, but we use this in UC: https://mlem.ai/doc/use-cases/dvc

Reply:
I need to check whether this actually breaks something or if it's ok to have.

Reply:
Ok, I checked and it doesn't seem to cause any problems.

Reply:
Yep, I still think it's better in the use case (which is not actually even a use case, but a tutorial if we apply the DVC.org terminology) to mention the link to the DVC install page. People can decide how to install it. Unless mlem depends on it (but them there should be something like mlem[dvc] or something)

Reply:
Ok, it's here https://github.com/iterative/mlem.ai/pull/132
Could you please review?

Reply:
done, I meant though that I would not include DVC as a dependency ... I would mention in the tutorial something like: `We assume that DVC is installed ... use this page to install it ...`

Reply:
Then UC won't work for those who don't have DVC installed. Alternatively, we could ask for DVC to be installed globally and remove it from `requirements.txt`. Although I'm not too fond of this - I would try to keep all installables inside the venv we create, so the user could remove it later. Maybe DVC case is an exception here, don't know.

Reply:
do we need to update mlem -> mlem[dvc-s3, dvc-azure]?

Reply:
and remove `dvc` itself?

and even in this case not a big fan of this tbh, we should have a way to install `dvc-core-s3` or something, so that CLI is not installed

not a good practice to have CLI tools as dependencies, it smells

Reply:
> My point is that it's rare you would include dvc as a requirement unless you use it as a library (Python API).

Ok, IIUC, we could keep this as is, cause MLEM calls DVC API and DVC should be installed with `pip`. We already state that DVC is installed from pip. As we install all other packages without `--user`, I think we should install DVC like this as well. Do I misunderstand something?

Reply:
Btw, from the other hand we have `fastapi` and `docker` installed with pip, although they may be installed as MLEM dependencies 🤔 Not sure what's the right approach here, but I think we should be consistent?

Reply:
@shcheklein, maybe I misunderstood you here? Do you maybe suggest to `pip install mlem dvc[azure,s3]` first and then `pip install -r requirements.txt` that have all other dependencies in it?

Reply:
> , I guess your last comment was about replacing dvc[azure,s3] with mlem[dvc,dvc-azure,dvc-s3], not about this discussion itself. Forgot to do that 😅

yes! sorry about the confusion

if fastapi / docker are used only by MLEM I would definitely install them via MLEM also!